### ISSUE TITLE
RecursionError: maximum recursion depth exceeded in comparison

### ISSUE THREAD

**Issue by jkobject** [broadinstitute/dalmatian#18]
Since upgrading to V0.0.10 I get this error message for all functions.
`RecursionError: maximum recursion depth exceeded in comparison`
on this workspace `<dalmatian.wmanager.WorkspaceManager rmc-rnaseq/St_Jude_RMS>`

Best,

**Comment by agraubert** [comment]
Could you provide a stacktrace for one of these errors? I cannot reproduce this issue

**Comment by jkobject** [comment]
```stacktrace
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
<ipython-input-176-681de50fd797> in <module>()
----> 1 wm.update_sample_set(setname+"_normals_XY",normalsXY.tolist())

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/base.py in update_sample_set(self, sample_set_id, sample_ids)
   1153     def update_sample_set(self, sample_set_id, sample_ids):
   1154         """Update or create a sample set"""
-> 1155         self.update_entity_set('sample', sample_set_id, sample_ids)
   1156 
   1157 

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/wmanager.py in call_with_lock(self, *args, **kwargs)
    149     def call_with_lock(self, *args, **kwargs):
    150         with self.lock:
--> 151             return func(self, *args, **kwargs)
    152     return call_with_lock
    153 

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/wmanager.py in update_entity_set(self, etype, set_id, member_ids)
   1040         )
   1041         if self.live:
-> 1042             try:
   1043                 result = setter()
   1044                 if not result:

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/base.py in hound(self)
     90         """
     91         if self.__hound is None:
---> 92             self.__hound = HoundClient(self.get_bucket_id(), credentials=self.__credentials, user_project=self.__user_project)
     93         return self.__hound
     94 

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/wmanager.py in get_bucket_id(self)
    681     def get_bucket_id(self):
    682         """Get the GCS bucket ID associated with the workspace"""
--> 683         metadata = self.firecloud_workspace
    684         if 'workspace' in metadata and 'bucketName' in metadata['workspace']:
    685             return metadata['workspace']['bucketName']

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/wmanager.py in call_with_lock(self, *args, **kwargs)
    149     def call_with_lock(self, *args, **kwargs):
    150         with self.lock:
--> 151             return func(self, *args, **kwargs)
    152     return call_with_lock
    153 

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/wmanager.py in call_using_cache(self, *args, **kwargs)
    195                 return self.cache[_key]
    196             self.fail(message) # Fail otherwise
--> 197 
    198         return call_using_cache
    199 

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/wmanager.py in firecloud_workspace(self)
    689     @_read_from_cache('configs')
    690     def list_configs(self):
--> 691         """List configurations in workspace"""
    692         return self.tentative_json(firecloud.api.list_workspace_configs(
    693             self.namespace,

~/anaconda2/envs/py36/lib/python3.6/site-packages/firecloud/api.py in get_workspace(namespace, workspace)
   1197     """
   1198     uri = "workspaces/{0}/{1}".format(namespace, workspace)
-> 1199     return __get(uri)
   1200 
   1201 def get_workspace_acl(namespace, workspace):

~/anaconda2/envs/py36/lib/python3.6/site-packages/firecloud/api.py in __get(methcall, headers, root_url, **kwargs)
     90     if not headers:
     91         headers = _fiss_agent_header()
---> 92     r = __SESSION.get(urljoin(root_url, methcall), headers=headers, **kwargs)
     93     if fcconfig.verbosity > 1:
     94         print('FISSFC call: %s' % r.url, file=sys.stderr)

~/anaconda2/envs/py36/lib/python3.6/site-packages/requests/sessions.py in get(self, url, **kwargs)
    544 
    545         kwargs.setdefault('allow_redirects', True)
--> 546         return self.request('GET', url, **kwargs)
    547 
    548     def options(self, url, **kwargs):

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/wmanager.py in _firecloud_api_timeout_wrapper(*args, **kwargs)
     73         **{
     74             **{'timeout': timeout_state.timeout},
---> 75             **kwargs
     76         }
     77     )

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/wmanager.py in _firecloud_api_timeout_wrapper(*args, **kwargs)
     76         }
     77     )
---> 78 
     79 # Monkey Patch the wrapped request method
     80 getattr(firecloud.api, "__SESSION").request = _firecloud_api_timeout_wrapper

... last 1 frames repeated, from the frame below ...

~/anaconda2/envs/py36/lib/python3.6/site-packages/dalmatian/wmanager.py in _firecloud_api_timeout_wrapper(*args, **kwargs)
     76         }
     77     )
---> 78 
     79 # Monkey Patch the wrapped request method
     80 getattr(firecloud.api, "__SESSION").request = _firecloud_api_timeout_wrapper

RecursionError: maximum recursion depth exceeded in comparison
```

**Comment by agraubert** [comment]
That's interesting. It seems like the wrapper that enforces timeouts is somehow stuck calling itself. I haven't encountered this issue, and when I force a stack trace, I'm only seeing one call to the timeout wrapper. Would you have some time today where I could come by your desk?

**Comment by jkobject** [comment]
yes feel free to do so today or tomorrow.

**Comment by agraubert** [comment]
We've identified the issue which causes the infinite recursion. The fix will be incorporated into dalmatian 0.0.12, which should be released later this week
